Blockmap storybook

Tishasoumya-02 opened the pull request.

Tishasoumya-02 committed:
MapsSidebar storybook

Tishasoumya-02 committed:
news

tiberiuichim reviewed (changes requested):
I think "Public components" is not the proper location for this story, maybe "Internal components/Maps/" ?

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

Tishasoumya-02 committed:
fix:towncrier

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

Tishasoumya-02 committed:
fix: storybook maps

Tishasoumya-02 commented:
@tiberiuichim @nileshgulia1 updated PR, please review!

stevepiercy commented on packages/volto/news/5090.feature:
```diff
@@ -0,0 +1 @@
+Storybook block/maps-@Tishasoumya
```
```suggestion
Add Storybook for maps block. @Tishasoumya
```

stevepiercy commented on packages/volto/src/components/manage/Blocks/Maps/MapsSidebar.stories.jsx:
```diff
@@ -0,0 +1,56 @@
+  argTypes: {
+    title: {
+      control: 'text',
+      description: 'Alt text of Map',
```
```suggestion
      description: 'Alternative text of the map block',
```

stevepiercy commented on packages/volto/src/components/manage/Blocks/Maps/MapsSidebar.stories.jsx:
```diff
@@ -0,0 +1,56 @@
+    },
+    url: {
+      control: 'text',
+      description: 'Url of the Map',
```
```suggestion
      description: 'URL of the map block',
```

stevepiercy reviewed (changes requested):
Minor tweaks to text strings. Still needs a technical review. Thank you!

Tishasoumya-02 committed:
Update packages/volto/news/5090.feature

Co-authored-by: Steve Piercy <web@stevepiercy.com>

Tishasoumya-02 committed:
Update packages/volto/src/components/manage/Blocks/Maps/MapsSidebar.stories.jsx

Co-authored-by: Steve Piercy <web@stevepiercy.com>

Tishasoumya-02 committed:
Update packages/volto/src/components/manage/Blocks/Maps/MapsSidebar.stories.jsx

Co-authored-by: Steve Piercy <web@stevepiercy.com>

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

stevepiercy reviewed (approved):
LGTM, but still needs a technical review from @plone/volto-team

stevepiercy commented:
Why does mr-roboto complain about the lack of a change log entry? It's in the right place.

Tishasoumya-02 committed:
update news

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

Tishasoumya-02 committed:
Update 5090.feature

sneridagh commented:
@wesleybl I agree, let's create one. Also, in VLT I'm adding a wrapper for blocks, that even pulls the `StyleWrapper` for the blocks. @Tishasoumya-02 maybe it would be useful to backport it.

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

Tishasoumya-02 committed:
fix:move the blocks to blocks/ in storybook

Tishasoumya-02 committed:
add blockwrapper

Tishasoumya-02 commented:
@wesleybl @sneridagh I moved this to Blocks/Maps. Also I added the BlockWrapper in packages/volto/src/stories as in VLT, if it's not the right place , where should I place this file? Also if some other feedbacks?

stevepiercy commented on packages/volto/news/5090.feature:
```diff
@@ -0,0 +1 @@
+Add Storybook for maps block @Tishasoumya-02
```
```suggestion
Add Storybook for maps block. @Tishasoumya-02
```

stevepiercy committed:
Update packages/volto/news/5090.feature

Tishasoumya-02 commented:
@stevepiercy, as far as I understood from there docs, these storybook features are mostly plug-and-play and don’t require much rewriting or refactoring to take advantage of. I could try a mini pilot to see how it fits. Some of these features are already used in VLT. Enabling MDX, autodocs and accessibility checks here could really improve how we document components and onboard others. We’d just need to make sure the right addons (like @storybook/addon-docs and @storybook/addon-a11y) are enabled — setup should be pretty straightforward.

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

Tishasoumya-02 committed:
Merge branch 'main' into blockmap-storybook

davisagli commented:
This is failing with an error on the MapsSidebar story ("onChangeBlock is not a function"). And we probably need a broader effort to be consistent about storybook pages for all blocks. I'll close this since it's old.
